modules/HapticGlove_module/src/GloveWearable.cpp

```diff
@@ -0,0 +1,434 @@
+
+    size_t i = 0;
+    for (const auto& sensor : m_fingertipLinkSensors)
+    {
+        if (sensor->getSensorStatus() != wearable::sensor::SensorStatus::Ok)
+        {
+            std::string sName = sensor->getSensorName();
+            yError() << m_logPrefix << "sensor status is not OK, sensor name: " << sName;
+
+            wearable::sensor::SensorStatus status = sensor->getSensorStatus();
+            switch (status)
+            {
+            case wearable::sensor::SensorStatus::Ok:
+                yInfo() << "sensor status: ok";
+                break;
+            case wearable::sensor::SensorStatus::Error:
+                yInfo() << "sensor status: Error";
+                break;
+            case wearable::sensor::SensorStatus::Calibrating:
+                yInfo() << "sensor status: Calibrating";
+                break;
+            case wearable::sensor::SensorStatus::Overflow:
+                yInfo() << "sensor status: Overflow";
+                break;
+            case wearable::sensor::SensorStatus::Timeout:
+                yInfo() << "sensor status: Timeout";
+                break;
+            case wearable::sensor::SensorStatus::Unknown:
+                yInfo() << "sensor status: Unknown";
+                break;
+            case wearable::sensor::SensorStatus::WaitingForFirstRead:
+                yInfo() << "sensor status: WaitingForFirstRead";
+                break;
+            default:
+                yInfo() << "default case!";
+                break;
+            }
+        }
+
+        wearable::Quaternion orientation;
+        wearable::Vector3 position;
+        sensor->getLinkPose(position, orientation);
+        for (size_t j = 0; j < 3; j++)
+            values(i, j) = position[j];
+        for (size_t j = 0; j < 4; j++)
+            values(i, j + 3) = orientation[j];
+        i++;
+    }
+    return true;
+}
+
+bool GloveWearableImpl::setFingertipForceFeedbackValues(const std::vector<int>& values)
+{
+    if (values.size() != m_numForceFeedback)
+    {
+        yError() << m_logPrefix
+                 << "size of the force feedback vector is not equal to the size of the default "
+                    "force feedback size.";
+    }
+    for (size_t i = 0; i < values.size(); i++)
+    {
+        std::string fingerName = m_humanFingerNameList[i];
+
+        wearable::msg::WearableActuatorCommand& wearableActuatorCommand
+            = m_iWearActuatorPort.prepare();
+
+        wearableActuatorCommand.value = values[i];
+        wearableActuatorCommand.info.name = m_wearablePrefix
+                                            + wearable::actuator::IHaptic::getPrefix() + fingerName
+                                            + "::ForceFeedback";
+        wearableActuatorCommand.info.type = wearable::msg::ActuatorType::HAPTIC;
+        wearableActuatorCommand.info.status = wearable::msg::ActuatorStatus::OK;
+        wearableActuatorCommand.duration = 0;
+
+        m_iWearActuatorPort.write(true);
```

Comment:
fixed.

Reply:
@S-Dafarra  @traversaro  @diegoferigo Following this comment, which we set the `writeStrict` to `false`, the effect is that, it is not correctly sending the haptic feedback for all the fingers. It only sends the thumb haptic feedback and the rest is ignored (to be more precise `HapticGlove::haptic::l_thumb_finger::ForceFeedback` which is the first one to set). Any suggestion? should I revert it back to `true` in the next PR, or there might be some other solutions!

Reply:
I have tested that with the real robot. Actually in simulation is the same, it is clear in the simulation video I putted here https://github.com/robotology/walking-teleoperation/pull/72#issuecomment-1014657236 , in the right-bottom terminal.

Reply:
I guess this is because every message you write in ``m_iWearActuatorPort`` have different meaning every time, and it is not just the same data sent over and over with a different value.

I guess this is one of the cases where ``forceStrict`` was indeed necessary. 

I think it is ok to open a PR even for just a line modification, it will be just easier to review :grin: